### PR TITLE
project Labyrinth by Anna Lindgren & Caroline Byström

### PR DESCRIPTION
**Pull request by Frivridna**
Link to my deployed project: https://labyrinth-by-caroline-anna.netlify.app/

Things I'd like to have clarified or explained in more detail: How to persist redux store to localStorage, so the app data remains upon reloading the page.

### CODE REVIEW

File: code/src/components/EndScreen.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  return (
+      <EndContainer>
+          <div class="nes-container is-rounded is-dark">
+              <Header>{lastDescription.description}</Header>
+              <OctoCat>
+              <i className="nes-octocat animate"></i>
+              </OctoCat>
+              <CssCredit>CSS from <a href="https://nostalgic-css.github.io/NES.css/">Ness.css</a> 
+              </CssCredit>
```
Comment (Sofarz) [inline code/src/components/EndScreen.js:16]: Check indentation here :)

File: code/src/components/GameScreen.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                  className="nes-btn"
+                  onClick={() => dispatch(CarryOnGame(action.direction))}>
+                      {action.direction}
+              </button>
+            </GameDescriptionContainer>
+          )
+        )}
+      <button
+        className="nes-btn"
+        disabled={!history.length}
```
Comment (Sofarz) [inline code/src/components/GameScreen.js:40]: The button is not disabled in the first description. Maybe try this instead: history.length < 1?
Right now you end up showing a blank box when pressing go back from the first description.

File: code/src/components/Games.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+const Games = () => {
+  const userName = useSelector(store => store.games.username)
+  const error = useSelector(store => store.games.error)
+  const loading = useSelector(store => store.games.loading)
+
+  return (
+    <MainWrapper>
+      <GameWrapper>
+      {error && `Ups, something went wrong, reason : ${error}`}
```
Comment (Sofarz) [inline code/src/components/Games.js:18]: Way to go on implementing an error message! 💯   Did you get it to work? Not sure how we can test it :)

File: code/src/components/Games.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      {loading && <Loader />}
```
Comment (Sofarz) [inline code/src/components/Games.js:19]: Really nice looking  loading spinner!

File: code/src/components/Games.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      {userName &&  <Header />}
+      {userName
+          ? <GameScreen />
+          : <StartScreen /> 
+      }
```
Comment (Sofarz) [inline code/src/components/Games.js:24]: Nice and clean structure!

File: code/src/components/Header.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+  return (
+    <HeaderContainer>
+        <i className="nes-icon is-medium star"></i>
+        <button className="nes-btn is-warning" onClick={onRestartButton}>
+            Restart
+        </button>
+        <i className="nes-icon is-medium star"></i>
+    </HeaderContainer>
+  )
```
Comment (Sofarz) [inline code/src/components/Header.js:18]: Maybe only have the restart button in the end page? Just a thought :)

File: code/src/components/Loader.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+import { Circular } from 'styled-loaders-react'
```
Comment (Sofarz) [inline code/src/components/Loader.js:3]: Nice 🥇

File: code/src/reducers/games.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+export const games  = createSlice({
```
Comment (Sofarz) [inline code/src/reducers/games.js:3]: You could do an export default of games instead to make it more clear in the components which one is the main slice.

**Review by Sofarz — APPROVED** [review]
Great job, the game looks awesome! 🌟  

We wrote some suggestions on minor changes that you could consider, but all in all very nice project. We liked that you used NES to style and the react loading spinner and that you implemented the error message! 
You completed the blue requirements and some of the red stretch goals, well done! 😃

File: code/src/components/GameScreen.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  const actions = useSelector(store => store.games.gameStatus.actions)
+  const history = useSelector(store => store.games.history)
+
+  const dispatch = useDispatch()
+
+  const onGoBack = () => {
+    dispatch(games.actions.setPreviousGameStatus())
+  }
+
+  if (actions?.length === 0) {
```
Comment (KarolCislowski) [inline code/src/components/GameScreen.js:19]: nice solution to determine if the game is finished.

**Review by KarolCislowski — APPROVED** [review]
Great App 🥳

**Pull request closed without merging**